modules/library/main/src/test/java/org/geotools/filter/function/QuantileFunctionTest.java

```diff
@@ -235,4 +237,89 @@ public void testConstantValuesString() {
+        assertEquals(percentages.length, 2);
+        assertEquals(percentages[0], 50.0);
+        assertEquals(percentages[1], 50.0);
+
+        // not numerical
+        SimpleFeatureType dataType =
+                DataUtilities.createType("classification.test1", "id:0,value:String");
+        String sVal[] = new String[] {"a", "b", "c", "d", "e", "f"};
+        SimpleFeature[] myfeatures = new SimpleFeature[sVal.length];
+        for (int i = 0; i < sVal.length; i++) {
+            myfeatures[i] =
+                    SimpleFeatureBuilder.build(
+                            dataType,
+                            new Object[] {Integer.valueOf(i + 1), sVal[i]},
+                            "classification.test1" + (i + 1));
+        }
+
+        SimpleFeatureCollection myFeatureCollection = DataUtilities.collection(myfeatures);
+        org.opengis.filter.expression.Expression func2 =
+                ff.function("Quantile", ff.property("value"), ff.literal(6), ff.literal(true));
+
+        Object value2 = func2.evaluate(myFeatureCollection);
+        // assertTrue(value2 instanceof RangedClassifier);
+        Classifier ranged2 = (Classifier) value2;
+        double[] percentages2 = ranged2.getPercentages();
+        assertEquals(percentages2.length, 6);
+        for (int i = 0; i < percentages2.length; i++) {
+            assertEquals(Math.floor(percentages2[i]), 16.0);
+        }
+    }
+
+    public void testPercentagesConsistencyWithClasses() throws SchemaException {
+        // create a feature collection with five features values 1-5
+        SimpleFeatureType dataType =
+                DataUtilities.createType("classification.test1", "id:0,value:int");
+        int iVal[] = new int[] {1, 1, 2, 3, 4, 5};
+        SimpleFeature[] myfeatures = new SimpleFeature[iVal.length];
+        for (int i = 0; i < iVal.length; i++) {
+            myfeatures[i] =
+                    SimpleFeatureBuilder.build(
+                            dataType,
+                            new Object[] {Integer.valueOf(i + 1), Integer.valueOf(iVal[i])},
+                            "classification.test1" + (i + 1));
+        }
+
+        // test unequally distributed classes
+        SimpleFeatureCollection myFeatureCollection = DataUtilities.collection(myfeatures);
+        org.opengis.filter.expression.Expression func2 =
+                ff.function("Quantile", ff.property("value"), ff.literal(5), ff.literal(true));
+
+        Object value2 = func2.evaluate(myFeatureCollection);
+        assertTrue(value2 instanceof RangedClassifier);
+        RangedClassifier ranged2 = (RangedClassifier) value2;
+        double[] percentages2 = ranged2.getPercentages();
+        assertEquals(percentages2.length, 5);
+        assertEquals(Math.floor(percentages2[0]), 33.0);
+        for (int i = 1; i < percentages2.length; i++) {
+            assertEquals(Math.floor(percentages2[i]), 16.0);
+        }
+
+        // test requested more classes than breaks
```

Comment:
Same as in the other PRs, can you have a test method run one test scenario instead of a bunch mixed toghether?

Reply:
fixed